src/MiniCover.HitServices/HitService.cs

```diff
@@ -40,11 +41,16 @@ public void Exit()
             {
                 if (clearTestMethodCache)
                 {
-                    this.Save();
+                    AppDomain.CurrentDomain.ProcessExit += CurrentDomain_ProcessExit;
```

Comment:
@bhugot 
I run some tests using project sample provided by @TheAngryByrd on issue https://github.com/lucaslorentz/minicover/issues/42 

He has a test with parallel executions, and I faced a deadlock if we try to write files at method exit.
I was thinking about it, adding locks to application methods might generate that.

I'm not sure how to solve it. For now if each MethodContext subscribe to ProcessExit, it seems that each MethodContext have 2 seconds to execute. But the problem is that we keep a lot of objects in memory.

Any ideas about avoiding deadlock?

Reply:
The problem is that you are going to write in same file. yes I  think I know how to sync the write process so that we don't lock and don't have to use processExit